### PR TITLE
chore: add integration tests for docker compose deployment

### PR DESCRIPTION
**Pull request by wojtekzyla**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix
- [ ] New feature
- [x] Refactor/improvement
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Checklist

- [x] My commit message is [conventional](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- [x] I have run pre-commit on all files before creating the PR
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -154,6 +155,7 @@ integration_tests/scripts/main.tf
 integration_tests/scripts/.terraform
 .DS_Store
 snmp-ssh-key.pem
+new_integration_testing
```
Comment (ajasnosz) [inline .gitignore:158]: is it created while running tests?

File: integration_tests/compose_deployment/splunk_test_utils_compose.py [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+        fi
+        sleep 1
+    done 
+    """
+    with open("check_for_containers.sh", "w") as fp:
+        fp.write(script_body)
+    os.system("chmod a+x check_for_containers.sh && ./check_for_containers.sh")
+
+
+# if __name__ == "__main__":
```
Comment (ajasnosz) [inline integration_tests/compose_deployment/splunk_test_utils_compose.py:156]: is this part for local test?

**Review by ajasnosz — CHANGES_REQUESTED** [review]
Would it be possible to parametrise the tests to not copy them and have in 2 separate folders?

File: .gitignore (hunk 1 above)
Comment (wojtekzyla) [reply .gitignore:158]: Oh sorry, `new_integration_testing` was a temporary directory, I'll delete this entry from .gitignore

File: integration_tests/compose_deployment/splunk_test_utils_compose.py (hunk 2 above)
Comment (wojtekzyla) [reply integration_tests/compose_deployment/splunk_test_utils_compose.py]: No, this is run in the pipeline. We just save this data to file and run this file. I copied this solution from the similar function used in tests for microk8s deployment https://github.com/splunk/splunk-connect-for-snmp/blob/main/integration_tests/splunk_test_utils.py#L180

**Review by ajasnosz — APPROVED** [review]

**Pull request merged**